vertexai/genai/option.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genai
+
+import (
+	"google.golang.org/api/option"
+	"google.golang.org/api/option/internaloption"
+)
+
+// WithUseREST is an option that may be passed to a Vertex AI client on
+// creation. If true, the client will use REST for transport; if false (default),
+// the client will use gRPC for transport.
+func WithUseREST(b bool) option.ClientOption {
```

Comment:
This could take no arguments: its mere presence would signal to use REST.
That gives us a path forward if there are other transports: WIthT1, WithT2, and so on.
(Of course we'd have to check that at most one was supplied).
What is the use case for `WithREST(false)`?